src/Microsoft.OData.Core/UriParser/Aggregation/AggregationMethod.cs

```diff
@@ -9,7 +9,31 @@ namespace Microsoft.OData.UriParser.Aggregation
     /// <summary>
     /// Enumeration of methods used in the aggregation clause
     /// </summary>
-    public enum AggregationMethod
+    public class AggregationMethod
+    {
+        public AggregationMethodType MethodType { get; private set; }
+        public string MethodLabel { get; private set; }
+
+        private AggregationMethod(AggregationMethodType aggregationMethodType)
+        {
+            this.MethodType = aggregationMethodType;
+        }
+
+        public static AggregationMethod Sum = new AggregationMethod(AggregationMethodType.Sum);

```

Comment:
Tried to, got some errors about the reference type being mutable. As we talked offline, won't do it.